guides/react_checkout_guide/11_finializing_on_the_client_side.md

```diff
@@ -0,0 +1,132 @@
+## Finalizing and back to the client
+
+Okay...The final stretch
+As always let's start with a new test.
+
+Add the following code to your feature test:
+
+```js
+it("user can finalize the order", () => {
+  cy.get("#product-2").within(() => {
+    cy.get("button").contains("Add to order").click();
+  });
+  cy.get("#product-3").within(() => {
+    cy.get("button").contains("Add to order").click();
+  });
+  cy.get("button").contains("View order").click();
+  cy.route({
+    method: "PUT",
+    url: "http://localhost:3000/api/orders/1",
+    response: { message: "Your order will be ready in 30 minutes!" },
+  });
+  cy.get("button").contains("Confirm!").click();
+  cy.get(".message").should(
+    "contain",
+    "Your order will be ready in 30 minutes!"
+  );
+});
+```
+
+We also want to make sure that we have som configuration to use in the cypress tests. We are adding a delay depending on the different functions that cypress runs.
+
+```js
+// cypress/support/commands.js
+const COMMAND_DELAY = 150;
+
+for (const command of [
+  "get",
+  "visit",
+  "click",
+  "trigger",
+  "type",
+  "clear",
+  "reload",
+  "contains",
+]) {
+  Cypress.Commands.overwrite(command, (originalFn, ...args) => {
+    const origVal = originalFn(...args);
+
+    return new Promise((resolve) => {
+      setTimeout(() => {
+        resolve(origVal);
+      }, COMMAND_DELAY);
+    });
+  });
+}
+```
+
+Let's work on the implementation code:
+
+Update the state object with orderTotal
+
+```js
+state = {
+  productData: [],
+  message: {},
+  orderDetails: {},
+  showOrder: false,
+  orderTotal: "",
+};
+```
+
+Add a new asynchronous function to handle the finalize order
+
+```js
+async finalizeOrder() {
```

Comment:
We need to convert all functions in this guide to arrow functions, it is what we expect of the students. We also can skip the binding of `this` when we call on these functions.